### PR TITLE
Set media_codecs based on target variant

### PR DESCRIPTION
**Pull request by k-mochi**
For different platform (CML/TGL), stagefright
service to parse media_codecs and media_codecs_performace
based on target variant at runtime to support single image.

Change-Id: Ie0ca43c5bce32b4146251a108f715fe0268be896
Tracked-On: OAM-97743
Signed-off-by: Kishan Mochi <kishan.mochi@intel.com>

### CODE REVIEW

File: groups/codecs/configurable/files.spec [hunk 1]
```diff
@@ -1,9 +1,14 @@
 [extrafiles]
 media_codecs_{{gpu}}.xml: "Specific configuration for audio and video codecs"
+media_codecs_gen9.xml: "Specific configuration for audio and video codecs for gen9"
```
Comment (TianmiChen) [inline groups/codecs/configurable/files.spec:3]: These files will be generated automatically while lunching the specific target. Do we need to add them here?

File: groups/codecs/configurable/files.spec [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+media_codecs_gen12.xml: "Specific configuration for audio and video codecs for gen12"
 mfx_omxil_core.conf: "MSDK configuration for video codecs"
 media_profiles.xml: "Media profile file"
 media_profiles_1080p.xml: "Media profile file with 1080p support"
 media_codecs_performance_{{platform}}.xml: "Media codecs performance file"
+media_codecs_performance_cml.xml: "Media codecs performance file for cml"
```
Comment (TianmiChen) [inline groups/codecs/configurable/files.spec:9]: Same as above.

File: groups/codecs/configurable/files.spec (hunk 1 above)
Comment (k-mochi) [reply groups/codecs/configurable/files.spec:3]: Regardless of what is specified in mixins.spec for target, we need these files for auto detect platform(CML/TGL) at runtime and load them. if target CPU model is not 166/141, what ever is specified in mixins.spec will be used.
i.e. media_codecs.xml/media_codecs_performance.xml

File: groups/codecs/configurable/files.spec (hunk 2 above)
Comment (dyang23) [reply groups/codecs/configurable/files.spec:9]: per offline discuss, this solution should works.

**Review by TianmiChen — APPROVED** [review]

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**